src/FhirLoader.Common/ProfileFileHandler.cs

```diff
@@ -0,0 +1,50 @@
+﻿using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace FhirLoader.Common
+{
+    public class ProfileFileHandler : FhirFileHandler
+    {
+        private readonly Stream _inputStream;
+        private readonly ILogger _logger;
+        private IEnumerable<ProcessedBundle>? _bundles;
+
+        public ProfileFileHandler(Stream inputStream, string fileName, int bundleSize, ILogger logger) : base(fileName, bundleSize)
+        {
+            _inputStream = inputStream;
+            _logger = logger;
+        }
+
+        public override IEnumerable<ProcessedBundle> FileAsBundles
+        {
+            get
+            {
+                if (_bundles is null)
+                    _bundles = ConvertToBundles();
+
+                return _bundles;
+            }
+        }
+
+        public IEnumerable<ProcessedBundle> ConvertToBundles()
+        {
+            JObject bundle;
+
+            // We must read the full file
+            using (StreamReader reader = new StreamReader(_inputStream))
+                bundle = JObject.Parse(reader.ReadToEnd());
+
+            yield return new ProcessedBundle
```

Comment:
Since this is just a bundle of 1, would be best to just return an object representing the resource without the bundle wrapper. This would make the code more clear.

Reply:
Do you mean to remove the processhandler and return the JObject or remove the yield? If we remove the processhandler, then we have to change the code from the program.cs file to the FhirResourceClient send method. Can you please elaborate it more?

Reply:
Actually I misread this code - it looks good! I just thought the `BundleText` was a resource of type `Bundle` but was just confused by the property names.